### PR TITLE
Move Session independent code from browser-session to feature-session.

### LINKED ISSUE #8304: Move LastAccessMiddleware
I saw that `LastAccessMiddleware` is in `mozilla.components.browser.session.engine.middleware` and gets added to `EngineMiddleware`. This is technically not correct. This has nothing to do with the `Engine` and should probably be moved somewhere else.

### LINKED ISSUE #9246: Move UndoMiddleware out of browser-session
Currently it needs `SessionManager`. But I think we can migrate that to using `TabsUseCases`.

### PR DESCRIPTION
**Pull request by pocmo**
Slowly thinning `browser-session`. :)

Also closes #8304. And closes #9246.

### CODE REVIEW

File: components/browser/session/src/main/java/mozilla/components/browser/session/SelectionAwareSessionObserver.kt [hunk 1]
```diff
@@ -1,76 +0,0 @@
...
- * specified or selected session, and can automatically take care of switching
- * over the observer in case a different session gets selected (see
- * [observeFixed] and [observeSelected]).
- *
- * @property activeSession the currently observed session
- * @property sessionManager the application's session manager
- */
-@Deprecated(OBSERVER_DEPRECATION_MESSAGE)
-@Suppress("DEPRECATION") // SessionManager observable is deprecated
-abstract class SelectionAwareSessionObserver(
```
Comment (csadilek) [inline components/browser/session/src/main/java/mozilla/components/browser/session/SelectionAwareSessionObserver.kt:22]: 👋

File: samples/browser/src/main/java/org/mozilla/samples/browser/DefaultComponents.kt [hunk 2]
```diff
@@ -141,7 +142,8 @@ open class DefaultComponents(private val applicationContext: Context) {
                 LocationService.default()
             ),
             SearchMiddleware(applicationContext),
-            RecordingDevicesMiddleware(applicationContext)
+            RecordingDevicesMiddleware(applicationContext),
+            LastAccessMiddleware()
```
Comment (csadilek) [inline samples/browser/src/main/java/org/mozilla/samples/browser/DefaultComponents.kt:146]: Should we put up a PR for Fenix for this? It's easy to miss I guess as it won't complain at compile-time :).

**Review by csadilek — APPROVED** [review]
✨

File: samples/browser/src/main/java/org/mozilla/samples/browser/DefaultComponents.kt (hunk 2 above)
Comment (pocmo) [reply samples/browser/src/main/java/org/mozilla/samples/browser/DefaultComponents.kt:146]: Yeah, I have it on my list. Will try to sync this up with an AC Nightly and update Fenix :)

**Pull request merged**